extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusMediatorConfiguration.java

```diff
@@ -0,0 +1,211 @@
+        return parameterTypes;
+    }
+
+    public void setParameterTypes(Class<?>[] parameterTypes) {
+        this.parameterTypes = parameterTypes;
+    }
+
+    public Shape getShape() {
+        return shape;
+    }
+
+    public void setShape(Shape shape) {
+        this.shape = shape;
+    }
+
+    @Override
+    public Shape shape() {
+        return shape;
+    }
+
+    @Override
+    public String getIncoming() {
+        return incoming;
+    }
+
+    public void setIncoming(String incoming) {
+        this.incoming = incoming;
+    }
+
+    @Override
+    public String getOutgoing() {
+        return outgoing;
+    }
+
+    public void setOutgoing(String outgoing) {
+        this.outgoing = outgoing;
+    }
+
+    @Override
+    public Acknowledgment.Strategy getAcknowledgment() {
+        return acknowledgment;
+    }
+
+    public void setAcknowledgment(Acknowledgment.Strategy acknowledgment) {
+        this.acknowledgment = acknowledgment;
+    }
+
+    public Integer getBroadcastValue() {
+        return broadcastValue;
+    }
+
+    public void setBroadcastValue(Integer broadcastValue) {
+        this.broadcastValue = broadcastValue;
+    }
+
+    @Override
+    public boolean getBroadcast() {
+        return broadcastValue != null;
+    }
+
+    public MediatorConfiguration.Production getProduction() {
+        return production;
+    }
+
+    public void setProduction(MediatorConfiguration.Production production) {
+        this.production = production;
+    }
+
+    @Override
+    public MediatorConfiguration.Production production() {
+        return production;
+    }
+
+    public MediatorConfiguration.Consumption getConsumption() {
+        return consumption;
+    }
+
+    public void setConsumption(MediatorConfiguration.Consumption consumption) {
+        this.consumption = consumption;
+    }
+
+    @Override
+    public MediatorConfiguration.Consumption consumption() {
+        return consumption;
+    }
+
+    public boolean isUseBuilderTypes() {
+        return useBuilderTypes;
+    }
+
+    public void setUseBuilderTypes(boolean useBuilderTypes) {
+        this.useBuilderTypes = useBuilderTypes;
+    }
+
+    @Override
+    public boolean usesBuilderTypes() {
+        return useBuilderTypes;
+    }
+
+    public Merge.Mode getMerge() {
+        return merge;
+    }
+
+    public void setMerge(Merge.Mode merge) {
+        this.merge = merge;
+    }
+
+    @Override
+    public Class<? extends Invoker> getInvokerClass() {
+        return invokerClass;
+    }
+
+    public void setInvokerClass(Class<? extends Invoker> invokerClass) {
+        this.invokerClass = invokerClass;
+    }
+
+    @Override
+    public String methodAsString() {
+        return getBean().getBeanClass().getName() + "#" + getMethodName();
+    }
+
+    @Override
+    public Method getMethod() {
+        throw new IllegalStateException("getMethod is not meant to be called on " + this.getClass().getName());
```

Comment:
Can it be an `UnsupportedOperationException`

Reply:
Done